python/paddle/fluid/framework.py

```diff
@@ -774,32 +774,84 @@ def attr(self, name):
         """
         return self.desc.attr(name)
 
+    def block_attr_id(self, name):
+        """
+        Get the block attribute's id by name.
+
+        Args:
+            name(str): the attribute name.
+
+        Returns:
+            (int): the block index.
+        """
+        return self.desc.block_attr_id(name)
+
     def block_attr(self, name):
         """
-        Get the block attribute by name.
+        Get the block attribute  by name.
+
+        Args:
+            name(str): the attribute name.
+
+        Returns:
+            (block): the block attribute.
+        """
+
+        id = self.block_attr_id(name)
+        assert (id >= 0 and id < len(self.block.program.blocks))
+        return self.block.program.blocks[id]
+
+    def blocks_attr(self, name):
+        """
+        Get the blocks attribute  by name.
 
         Args:
             name(str): the attribute name.
 
         Returns:
-            int: the block index.
+            attrs(list[block]): the blocks attribute.
         """
-        return self.desc.block_attr(name)
+        attrs = []
+        for i in self.blocks_attr_ids(name):
+            assert (i >= 0 and i < len(self.block.program.blocks))
+            attrs.append(self.block.program.blocks[i])
+
+        return attrs
+
+    def blocks_attr_ids(self, name):
+        """
+        Get the blocks attribute's ids by name.
+
+        Args:
+            name(str): the attribute name.
+
+        Returns:
+            (list[int]): the blocks index.
+        """
+
+        return self.desc.blocks_attr_ids(name)
 
     def all_attrs(self):
         """
         Get the attribute dict.
 
         Returns:
-            dict: The Operator's attribute dict.
+            attr_map(dict(name->attr)): The Operator's attribute dict.
```

Comment:
See: http://www.sphinx-doc.org/en/stable/ext/example_google.html docstring return comments better to be like the format:

```
    Returns:
    type: comment
```

Reply:
Don.Thanks!